session.go

```diff
@@ -151,6 +152,8 @@ type session struct {
 	receivedFirstPacket bool
 
 	sessionCreationTime time.Time
+
+	mutex sync.Mutex // protects lastPacketReceivedTime
```

Comment:
The mutex will also need to be locked wherever lastPacketReceivedTime is read. It might be easier to just turn it into a function?